packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -322,4 +333,56 @@ abstract contract LivenessModule2 {
         delete challengeStartTime[_safe];
         emit ChallengeCancelled(_safe);
     }
+
+    /// @notice Internal function to clear the liveness module configuration and any active challenge.
+    /// @param _safe The Safe instance to clear the configuration for.
+    function _clearLivenessModule(Safe _safe) internal {
+        // Erase the configuration data for this safe
+        delete livenessSafeConfiguration[address(_safe)];
+        // Also clear any active challenge
+        _cancelChallenge(address(_safe));
+        emit ModuleCleared(address(_safe));
+    }
+
+    /// @notice Internal function to disable this guard from the given Safe.
+    /// @dev Only disables the guard if it is enabled, otherwise does nothing in case another
+    ///      guard is enabled.
+    /// @param _targetSafe The Safe instance to disable this guard from.
+    function _disableAndClearGuard(Safe _targetSafe) internal virtual;
+
+    /// @notice Internal function to disable this module from the given Safe.
+    /// @param _targetSafe The Safe instance to disable this module from.
+    function _disableAndClearThisModule(Safe _targetSafe) internal {
+        // Get current modules
+        // This might not work if you have more than 100 modules, but what are you even doing if that's the case?
```

Comment:
```suggestion
        // This might not do anything if you have more than 100 modules, but what are you even doing if that's the case?
        // Even in that case, it wouldn't revert, which would be the real problem.
```